cpp/oneapi/dal/algo/decision_forest/backend/gpu/train_splitter_sp_opt_impl_dpc.cpp

```diff
@@ -162,6 +163,28 @@ sycl::event train_best_split_sp_opt_impl<Float, Bin, Index, Task, sbg_size>::
     ONEDAL_ASSERT(device_has_enough_local_mem(queue, local_buf_byte_size));
     // TODO: add separate branch to process situation when there isn't enough local mem
 
+    // Random splitting helpers
+    auto ftr_random_select = pr::ndarray<Index, 1>::empty({ selected_ftr_count });
+    auto ftr_random_select_ptr = ftr_random_select.get_mutable_data();
+
+    if (ctx.splitter_mode_value == splitter_mode::random) {
+        pr::engine_collection collection(1, ctx.seed);
+        std::vector<pr::engine> engine_arr = collection([&](std::size_t i, std::size_t& j) {
+            j = i;
+        });
+        pr::rng<Index> rn_gen;
+        rn_gen.uniform( // Select bin treshold randomly
+            selected_ftr_count,
+            ftr_random_select_ptr,
+            engine_arr[0].get_state(),
+            0,
+            index_max);
+    }
+
+    auto ftr_rnd_trshld = ftr_random_select.to_device(queue);
+    auto ft_rnd_ptr = ftr_rnd_trshld.get_data();
+    // Random splitting helpers end
+
```

Comment:
It seems the cgh.parallel_for below (in the part that is left unchanged) should be modified in case of extra trees. Because we do not need parallelism across the bins of the histogram there. As extra trees only make 1 random split for each feature - no need this level of parallelism.